semgrep/semgrep/rule.py

```diff
@@ -219,7 +219,8 @@ def should_run_on_semgrep_core(self) -> bool:
 
         def has_runnable_rule(d: Dict[str, Any]) -> bool:
             for k in d:
-                if k in RuleValidation.PATTERN_KEYS:
+                # `pattern-inside` is valid without an accompanying `pattern`
```

Comment:
I'm not sure we want to accept this. Usually people use pattern-inside: with a pattern:.
If you just have pattern-inside: everywhere, you can simply use pattern: instead.

Reply:
You're right, but semgrep currently accepts this, so it would be weird if `project-depends-on` rules had different behavior.